tests/span_tests.cpp

```diff
@@ -204,6 +204,24 @@ SUITE(span_tests)
             auto workaround_macro = [=]() { span<int> s{p, 2}; };
             CHECK_THROW(workaround_macro(), fail_fast);
         }
+
+        {
+            auto s = make_span(&arr[0], 2);
+            CHECK(s.length() == 2 && s.data() == &arr[0]);
+            CHECK(s[0] == 1 && s[1] == 2);
+        }
+
+        {
+            int* p = nullptr;
+            auto s = make_span(p, static_cast<span<int>::index_type>(0));
```

Comment:
possibly, but I'm doing the same thing the non make_span test is doing to be consistent. To create the tests, I just cut / past all of the tests that apply and changed them to make_span to be sure that we hit everything